### PR TITLE
Expose embeddedObject.getParent

### PR DESCRIPTION
**Pull request by nirinchev**
This is a follow up on the embedded objects PR that I pulled to a separate branch because it depends on https://github.com/realm/realm-core/pull/5960.

### CODE REVIEW

**Review by desistefanova — COMMENTED** [review]
We have to update the core version in dart Changelog.

**Review by desistefanova — COMMENTED** [review]
Do you think having these new functions can help us to implement `DynamicRealm` `find` and `all` to support embedded objects?
Or it must be similar to Realm class and to support only RealmObjects?

**Comment by nirinchev** [comment]
They should be equivalent - i.e. we should not allow users to lookup embedded objects directly.

File: lib/src/realm_object.dart [hunk 1]
```diff
@@ -411,7 +411,17 @@ mixin RealmObjectBase on RealmEntity implements Finalizable {
 mixin RealmObject on RealmObjectBase {}
 
 /// @nodoc
-mixin EmbeddedObject on RealmObjectBase {}
+mixin EmbeddedObject on RealmObjectBase {
+  RealmObjectBase? getParent() {
```
Comment (nielsenko) [inline lib/src/realm_object.dart:415]: Should we make this a `parent` getter instead?

Comment (blagoev) [reply lib/src/realm_object.dart:415]: +1

Comment (blagoev) [inline lib/src/realm_object.dart:415]: no api doc

File: lib/src/util.dart [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Tuple<T1, T2> {
```
Comment (blagoev) [inline lib/src/util.dart:1]: we could move this to realm_common_base.dart and delete the utils.dart

File: lib/src/native/realm_core.dart [hunk 3]
```diff
@@ -1054,6 +1067,10 @@ class _RealmCore {
     return _realmLib.realm_equals(first._pointer.cast(), second._pointer.cast());
   }
 
+  int getObjectKey(RealmObjectBase obj) {
```
Comment (blagoev) [inline lib/src/native/realm_core.dart:1070]: is this used somewhere ?

**Review by blagoev — COMMENTED** [review]
It's ok but I have the feeling we missed something while rebasing. Since this `getObjectKey` does not seem to be used anywhere

**Comment by blagoev** [comment]
should have a changelog entry

**Review by blagoev — CHANGES_REQUESTED** [review]
can we check the rebase and also add a changelog entry. Just making it more visible here

File: lib/src/realm_object.dart (hunk 1 above)
Comment (nirinchev) [reply lib/src/realm_object.dart:415]: I don't think it's a good idea, because users may want to use `parent` as a property on their models.

Comment (nielsenko) [reply lib/src/realm_object.dart:415]: Had a long discussion about this with @nirinchev and we settled on trying with a getter

Comment (nielsenko) [reply lib/src/realm_object.dart]: Added api doc on `getParent` now `get parent`

File: lib/src/util.dart (hunk 2 above)
Comment (nielsenko) [reply lib/src/util.dart:1]: Done

**Review by blagoev — APPROVED** [review]

**Review by desistefanova — APPROVED** [review]

**Review by nielsenko — APPROVED** [review]

**Pull request merged**